### PR TITLE
Streams inplace

### PR DESCRIPTION
**Pull request by AndreyTepaykin**

### CODE REVIEW

**Review by EGreg — CHANGES_REQUESTED** [review]
Whatever you change in Q.getter in web/Q.js also change it in classes/Q.js

File: platform/plugins/Streams/text/Streams/content/en.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
     "errors": {
-        "HiddenFromYou": "This content is hidden from you."
+        "HiddenFromYou": "This content is hidden from you.",
+        "StreamDoesntExists": "stream don't exists",
```
Comment (EGreg) [inline platform/plugins/Streams/text/Streams/content/en.json]: it should be StreamDoesntExist not "exists" with value "This stream doesn't exist", and we already have error "MissingStream" so we should have "Stream is missing." instead.

File: platform/plugins/Streams/text/Streams/content/en.json [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+        "NotEnoughPermissionsHandleStream": "not enough permissions to handle stream",
```
Comment (EGreg) [inline platform/plugins/Streams/text/Streams/content/en.json]: What is wrong to throw "Users_NotAuthorized" error?

File: platform/plugins/Streams/handlers/Streams/inplace/tool.php [hunk 3]
```diff
@@ -61,42 +56,61 @@ function Streams_inplace_tool($options)
...
+					'fallback' => $texts["errors"]["InplaceTypeMustBe"]
+				);
+		}
+		if (!$stream->testWriteLevel('suggest')) {
+			$toolOptions = array(
+				'fallback' => $texts["errors"]["NotEnoughPermissionsHandleStream"]
+			);
+		}
 	} else {
-		$field = !empty($options['field']) ? $options['field'] : 'content';
```
Comment (EGreg) [inline platform/plugins/Streams/handlers/Streams/inplace/tool.php:69]: Seems you deleted a lot of code, can you summarize what you did here?

**Review by EGreg — CHANGES_REQUESTED** [review]
Whatever you change in Q.getter in web/Q.js also change it in classes/Q.js

File: platform/plugins/Streams/handlers/Streams/inplace/tool.php (hunk 3 above)
Comment (AndreyTepaykin) [reply platform/plugins/Streams/handlers/Streams/inplace/tool.php:69]: Please look demo video

File: platform/plugins/Streams/web/js/tools/inplace.js [hunk 4]
```diff
@@ -122,114 +127,135 @@ Q.Tool.define("Streams/inplace", function (options) {
...
-				case 'select':
+
+		// dynamically construct the tool
+		var ipo = Q.extend(state.inplace, {
+			action: stream && stream.actionUrl(),
+			method: 'put',
+			field: state.field,
+			type: state.inplaceType
+		});
+		var value = state.fallback || (state.attribute ? stream.getAttribute(state.attribute) : stream.fields[state.field]) || "";
```
Comment (EGreg) [inline platform/plugins/Streams/web/js/tools/inplace.js]: You got it backwards ... value is supposed to be `(stream && (state.attribute ? stream.getAttribute(state.attribute) : stream.fields[state.field])) || state.fallback || ""`

**Review by EGreg — CHANGES_REQUESTED** [review]

File: platform/plugins/Streams/handlers/Streams/inplace/tool.php (hunk 3 above)
Comment (AndreyTepaykin) [reply platform/plugins/Streams/handlers/Streams/inplace/tool.php:69]: Explanation about removed code: https://drive.google.com/file/d/1lwS1KI8Kxly27ldewfTfB9VcOrZjmYFq/view?usp=sharing

File: platform/plugins/Streams/text/Streams/content/en.json (hunk 1 above)
Comment (AndreyTepaykin) [reply platform/plugins/Streams/text/Streams/content/en.json]: Use error message "Streams/inplace tool: stream is undefined" as tool content when 
publisherId or streamName undefined. This message used before, but as Exception.
So the text "StreamDoesntExists" don't need.
Commit: https://github.com/Qbix/Platform/pull/1933/commits/e9172bb862712a20c1996e4df290f5e869ef2e86

File: platform/plugins/Streams/text/Streams/content/en.json (hunk 2 above)
Comment (AndreyTepaykin) [reply platform/plugins/Streams/text/Streams/content/en.json]: Removed checking $stream->testWriteLevel('suggest') on server side. Because first of 
all this check only necessary if state.editable==true, but second (and most important) 
this checked on client side, and if state.editable==true but logged in user have no permissions, 
inplace rendered as non editable, i.e. just display stream content in tool container.
Commit: https://github.com/Qbix/Platform/pull/1933/commits/63a2f2aa7bd2f8d63cc55a51b5aa1f54a29608dc

File: platform/plugins/Streams/web/js/tools/inplace.js (hunk 4 above)
Comment (AndreyTepaykin) [reply platform/plugins/Streams/web/js/tools/inplace.js]: Fixed value definition. First check if stream exists.
Commit: https://github.com/Qbix/Platform/pull/1933/commits/4c4f6bd14edb62afd22a7c8f8fc665fa62982c75

**Review by EGreg — APPROVED** [review]
